clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -291,14 +294,28 @@ private KafkaProducer(ProducerConfig config, Serializer<K> keySerializer, Serial
             } else {
                 this.requestTimeoutMs = config.getInt(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG);
             }
+            boolean idempotenceEnabled = config.getBoolean(ProducerConfig.IDEMPOTENCE_ENABLED_CONFIG);
+
+            if (idempotenceEnabled) {
+                this.transactionState = new TransactionState(idempotenceEnabled);
+                if (config.getInt(ProducerConfig.RETRIES_CONFIG) == 0) {
```

Comment:
I think raising an exception is better since it guarantees that the user will see the expectation clearly. Besides, what would we override the value with? MAX_VALUE?